lib/rust/parser/jni/src/lib.rs

```diff
@@ -72,6 +73,36 @@ pub extern "system" fn Java_org_enso_syntax2_Parser_parseInput(
     result.unwrap().into_inner()
 }
 
+/// Determine the token variant of the provided input.
+#[allow(unsafe_code)]
+#[no_mangle]
+pub extern "system" fn Java_org_enso_syntax2_Parser_isIdentOrOperator(
```

Comment:
This method is also needed here: https://github.com/enso-org/enso/pull/8779#discussion_r1467903242
- from code clarity perspective - we should have just a single implementation
- from performance perspective - it is better to stay in the JVM than jump out and back just to compare letters in a string - the overhead of `input` buffer conversion isn't negliable 
   - not only a JVM vs. JNI problem
   - also JavaScript vs. WASM problem

However if the performance wasn't a problem so far, we probably don't need to care. It is more important to get the functionality right.